### PR TITLE
Fix version issue of storage and meta binary

### PR DESCRIPTION
**Pull request by yixinglu**
reference vesoft-inc/nebula-common#372 vesoft-inc/nebula-common#373

image

### CODE REVIEW

File: package/package.sh [hunk 1]
```diff
@@ -90,6 +91,7 @@ function build {
     cmake \
         -DCMAKE_BUILD_TYPE=${build_type} \
         -DNEBULA_BUILD_VERSION=${version} \
+        -DGIT_INFO_SHA=${git_info_sha} \
```
Comment (laura-ding) [inline package/package.sh:94]: why need input?

File: package/package.sh [hunk 2]
```diff
@@ -79,6 +79,7 @@ function build {
     build_type=$3
     branch=$4
     build_dir=$PROJECT_DIR/build
+    git_info_sha=$(git rev-parse --short HEAD)
```
Comment (laura-ding) [inline package/package.sh:82]: cmakelist has do it

**Review by panda-sheep — APPROVED** [review]

**Comment by critical27** [comment]
`show hosts meta/graph/storage` previously will print the git info sha, so it won't work if we don't package?

**Comment by yixinglu** [comment]
> `show hosts meta/graph/storage` previously will print the git info sha, so it won't work if we don't package?
> `--version`? BTW, there is another http will show the sha.

previous git sha is common commit hash, not the storage repo commit info, since the cmake target defined in common repo. And if you want to embed the git sha into storage binary locally, you can modify the cmake configure options like `cmake -DGIT_INFO_SHA=$(git rev-parse --short HEAD) ...` to do that.

**Comment by yixinglu** [comment]
Hold on this PR, I will check whether `NEBULA_STRINGFY(GIT_INFO_SHA)` is expected when not define GIT_INFO_SHA macro.

**Comment by critical27** [comment]
> Hold on this PR, I will check whether `NEBULA_STRINGFY(GIT_INFO_SHA)` is expected when not define GIT_INFO_SHA macro.

👌

**Comment by critical27** [comment]
LGTM, we'd better add `DGIT_INFO_SHA` when package. Otherwise, storage and services will only print the `NEBULA_BUILD_VERSION`, which is set as graph SHA when packaging.

**Review by critical27 — DISMISSED** [review]

File: src/version/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,15 @@
+# Copyright (c) 2018 vesoft inc. All rights reserved.
```
Comment (laura-ding) [inline src/version/CMakeLists.txt:1]: 2020

File: src/version/Version.cpp.in [hunk 4]
```diff
@@ -0,0 +1,31 @@
+/* Copyright (c) 2018 vesoft inc. All rights reserved.
```
Comment (laura-ding) [inline src/version/Version.cpp.in:1]: 2020

File: src/CMakeLists.txt [hunk 5]
```diff
@@ -1,3 +1,8 @@
+# Copyright (c) 2018 vesoft inc. All rights reserved.
```
Comment (laura-ding) [inline src/CMakeLists.txt:1]: 2020

File: src/version/CMakeLists.txt (hunk 3 above)
Comment (yixinglu) [reply src/version/CMakeLists.txt]: fixed

File: src/CMakeLists.txt (hunk 5 above)
Comment (yixinglu) [reply src/CMakeLists.txt:1]: fixed

File: src/version/Version.cpp.in (hunk 4 above)
Comment (yixinglu) [reply src/version/Version.cpp.in]: fixed

**Review by laura-ding — APPROVED** [review]

File: src/version/Version.cpp.in [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+
+#include <folly/String.h>
+
+#cmakedefine NEBULA_BUILD_VERSION
+
+namespace nebula {
+namespace storage {
+
+std::string gitInfoSha() {
+    return "@GIT_INFO_SHA@";
```
Comment (critical27) [inline src/version/Version.cpp.in:17]: 👍

**Review by critical27 — APPROVED** [review]
Good job, thx!

**Pull request merged**